aikau/src/main/resources/alfresco/dnd/DroppedNestingItemWrapper.js

```diff
@@ -114,34 +133,142 @@ define(["dojo/_base/declare",
+                  this.alfLog("warn", "The 'targetWidget' has no 'targetProperty' or no 'getValue' function", evt.targetWidget, this);
+               }
             }
             else
             {
-               this.alfLog("warn", "The 'targetWidget' has no 'targetProperty' or no 'getValue' function", evt.targetWidget, this);
+               this.alfLog("warn", "The update event contains no 'targetWidget' attribute", evt, this);
             }
          }
+      },
+
+      /**
+       * Handles items in a target being reordered
+       * 
+       * @instance
+       * @param {object} evt The drop event.
+       */
+      onNestedTargetReordered: function alfresco_dnd_DroppedNestingItemWrapper__onNestedTargetReordered(evt) {
+         if (evt.targetWidget === this)
+         {
+            // We've managed to catch our own emitted event, allow bubbling to continue...
+         }
          else
          {
-            this.alfLog("warn", "The update event contains no 'targetWidget' attribute", evt, this);
+            if (typeof evt.stopPropagation === "function")
+            {
+               evt.stopPropagation();
+            }
+            if (typeof evt.preventDefault === "function")
+            {
+               evt.preventDefault();
+            }
+
+            // Attempt to mix the updated target value into the wrapped value...
+            if (evt.targetWidget && evt.targetWidget.targetProperty)
+            {
+               var listToReorder = lang.getObject(evt.targetWidget.targetProperty, false, this.value);
+               if (!ObjectTypeUtils.isArray(listToReorder))
+               {
+                  this.alfLog("warn", "Can't reorder because the targetProperty does not exist or is not an array", evt.targetWidget.targetProperty, this.value, this);
+               }
+               else if (isNaN(evt.oldIndex) || isNaN(evt.newIndex))
+               {
+                  this.alfLog("warn", "Can't reorder because either the oldIndex or the newIndex (or both) are missing or are not numbers", evt, this);
+               }
+               else
+               {
+                  // Do the re-ordering...
+                  var tmp1 = listToReorder[evt.oldIndex];
+                  var tmp2 = listToReorder[evt.newIndex];
+                  listToReorder[evt.oldIndex] = tmp2;
+                  listToReorder[evt.newIndex] = tmp1;
+
+                  // ...then emit an event to capture the new value...
+                  this.onUpdateParentOfChange();
+               }
+            }
+            else
+            {
+               this.alfLog("warn", "The update event contains no 'targetWidget' attribute", evt, this);
+            }
          }
       },
 
       /**
+       * This function can be called to update wrapping widgets that the current item has changed.
+       *
+       * @instance
+       */
+      onUpdateParentOfChange: function alfresco_dnd_DroppedNestingItemWrapper__onUpdateParentOfChange() {
```

Comment:
This is not an ideally-named method. The "on" prefix suggests that an event has just happened and this is the handler, but in fact this is the method that triggers the event. A more suitable name might be "updateParentOfChange", or even more accurately "notifyParentOfChange".